src/main/scala/com/example/MyRegistryActor.scala

```diff
@@ -0,0 +1,20 @@
+package com.example
+
+import akka.actor.{ Actor, ActorLogging, Props }
+
+object MyRegistryActor {
```

Comment:
its overkill to use actors, we just need function `String => Future[String]`, which in our case is super simple `(key: String) => Future(localStore.get(key))`

But if you insist on having a separate file for lookup, then it would look something like this
```
trait LookupFromRegistry extends OKTable with ((String) => Future[String]) {

  implicit val ex: ExecutionContext

  override def apply(key: String): Future[String] = Future(localStore.get(key))
}
```

Reply:
Yea I agree with Actors being overkill here.

WRT `trait ... extends ... with ((String) => Future[String])`, what is the purpose of extending my Trait with this Function type?

Reply:
so you can do 
``` 
    val lookup = new LookupFromRegistry {
       val ex = ExecutionContext.fromExecutor(Executors.newFixedThreadPool(32))
    }
```

remember `lookup` takes a function